From 2cbf9244f21d741929fd39b023328afe63e7988b Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Tue, 25 Sep 2018 16:22:46 +0800
Subject: [PATCH 4729/5242] MLK-19665: ARM64: dts: disable hdmi sound card if
 hdmi device is disabled

commit  bddd330b01192a0a986afb50b0edc0712953994c from
https://source.codeaurora.org/external/imx/linux-imx.git

Disable hdmi sound card if hdmi device is disabled, otherwise the sound
card will return retry (-517) error for the codec devices is not registered.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dtsi |    6 ++++++
 .../dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi |    6 ++++++
 .../freescale/fsl-imx8mq-evk-lcdif-adv7535.dtsi    |    4 ++++
 .../freescale/fsl-imx8mq-evk-lcdif-rm67191.dtsi    |    4 ++++
 4 files changed, 20 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dtsi
index ac41f65..640674f 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dtsi
@@ -12,6 +12,12 @@
  * GNU General Public License for more details.
  */
 
+/ {
+	sound-hdmi {
+		status = "disabled";
+	};
+};
+
 &hdmi {
 	status = "disabled";
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi
index 4a57d5a..9215b6d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-rm67191.dtsi
@@ -12,6 +12,12 @@
  * GNU General Public License for more details.
  */
 
+/ {
+	sound-hdmi {
+		status = "disabled";
+	};
+};
+
 &hdmi {
 	status = "disabled";
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dtsi
index 6a9be35..afe53f1 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dtsi
@@ -16,6 +16,10 @@
 	display-subsystem {
 		status = "disabled";
 	};
+
+	sound-hdmi {
+		status = "disabled";
+	};
 };
 
 &irqsteer_dcss {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-rm67191.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-rm67191.dtsi
index 27907fc..3b1a115 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-rm67191.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-rm67191.dtsi
@@ -16,6 +16,10 @@
 	display-subsystem {
 		status = "disabled";
 	};
+
+	sound-hdmi {
+		status = "disabled";
+	};
 };
 
 &irqsteer_dcss {
-- 
1.7.9.5

